File: backend/src/main/java/com/example/demo/service/AuthService.java
package com.example.demo.service;

import com.example.demo.dto.AuthResponse;
import com.example.demo.dto.UserResponse;

public interface AuthService {
    AuthResponse login(String username, String password);

    void logout(String token);

    UserResponse register(String username, String password);

    /**
     * Refresh access token
     * @return newAccessToken
     */
    String refreshToken(String refreshToken);

    UserResponse verifyToken(String token);
}

File: frontend/src/services/authService.ts
import axiosIns from "@/lib/axios";
import type { AuthResponse, RefreshResponse } from "@/lib/types";

const authService = {
  signin: (username: string, password: string) => {
    return axiosIns.post<AuthResponse>("/auth/login", { username, password });
  },
  refreshToken: () => {
    return axiosIns.post<RefreshResponse>("/auth/refresh");
  },
};

export default authService;
